storage/teststore/store.go

```diff
@@ -48,10 +45,22 @@ func (store *Client) indexOf(key storage.Key) (int, bool) {
 	return i, store.Items[i].Key.Equal(key)
 }
 
+func (store *Client) forcedError() bool {
+	if store.ForceError > 0 {
+		store.ForceError--
+		return true
+	}
+	return false
+}
+
 // Put adds a value to store
 func (store *Client) Put(key storage.Key, value storage.Value) error {
 	store.version++
 	store.CallCount.Put++
+	if store.forcedError() {
+		return errors.New("internal error")
```

Comment:
looks like you're calling internal error several times, should we extrapolate that ?